### ISSUE TITLE
 Abnormal(Impossible?)  increase in container size.

### ISSUE THREAD

**Issue by nickmaleao** [docker/docker#19584]
Hello

I have created a base container, with the purpose of being able to ssh to it, using the credentials of a windows ad server.

In the course of this message, I will be using the ad term to describe , a windows ad(Active Directory) server .

For this to work, I used a ubuntu image(LTS) and installed the following main packages, sssd, opensshserver, samba.

The container has 2 interfaces, one is connect to the default bridge docker0(172.17.0.1), and the other is connect to a 2º bridge interface (1.1.1.2) added thought the pipework  script.

After all is setup up, this is container state/size:

root@thanos:/home/domain.com/leonarit# docker ps -s
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES               SIZE
5d26b12ae2cf        04fb8b00035c        "/sbin/my_init"     About an hour ago   Up About an hour                        nms                 18.53 MB (virtual 1.707 GB)

If i ssh to the container using a local account, everything works normal, and once inside, i can change to the ad user, for example: “su leonarit”, which will use the ad server 1.1.1.10 as source off the accounting mechanism.

The problem only happens when i try to ssh with the ad credentials, after a successful ssh with the ad credentials, the container increases the an impossible size(292.3 GB (virtual 294 GB)), i dont even have that much space in my partition.

oot@thanos:/home/domain.com/leonarit# ssh leonarit@1.1.1.1
leonarit@1.1.1.1's password: 
Welcome to Ubuntu 14.04.2 LTS (GNU/Linux 3.16.0-59-generic x86_64)

BANNER

leonarit@domain.com@nms:/$ 
leonarit@domain.com@nms:/$ exit
logout
Connection to 1.1.1.1 closed.

root@thanos:/home/domain.com/leonarit# docker ps -s
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES               SIZE
5d26b12ae2cf        04fb8b00035c        "/sbin/my_init"     About an hour ago   Up About an hour                        nms                 292.3 GB (virtual 294 GB)

Host information: 

root@thanos:/home/domain.com/leonarit# docker info
Containers: 1
Images: 20
Server Version: 1.10.0-dev
Storage Driver: devicemapper
 Pool Name: docker-252:0-131076-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 107.4 GB
 Backing Filesystem: extfs
 Data file: /dev/ubuntu01-vg/data
 Metadata file: /dev/ubuntu01-vg/metadata
 Data Space Used: 3.854 GB
 Data Space Total: 46.98 GB
 Data Space Available: 43.13 GB
 Metadata Space Used: 3.572 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.144 GB
 Udev Sync Supported: false
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Library Version: 1.02.82-git (2013-10-04)
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.16.0-59-generic
Operating System: Ubuntu 14.04.3 LTS
CPUs: 4
Total Memory: 3.854 GiB
Name: thanos.lab16916.com
ID: 4YZK:VF2G:YKWJ:GN54:33FF:KRAY:WUJ4:FLRU:3T4C:FKVP:CNJE:KGVU
Debug mode (server): true
 File Descriptors: 18
 Goroutines: 36
 System Time: 2016-01-20T14:25:47.883054752Z
 EventsListeners: 0
 Init SHA1: 
 Init Path: /usr/bin/docker
 Docker Root Dir: /var/lib/docker/0.0
WARNING: No swap limit support
Experimental: true

I have tested Docker stable and experimental, ant the result is the same. I begun the test with the default store driver: aufs, and changed to devicemapper, but the problems remained the same.

Is there any way to do a debug, to try to figure out ,why Docker increases the size of the container beyond the possible space limit.

**Comment by HackToday** [comment]
Is it really that container take such space ?  

what's the container do? Is it possible to check logs in container and also enable debug in daemon docker ?

With logs it may show some helpful info

**Comment by nickmaleao** [comment]
I started docker as a daemon with the debug option, and after i ssh to the  container(nms),the size changes, but  nothing appears in the logs. I have logs from the startup of the container but nothing appears when i ssh,

I have checked the following locations:

/var/log/upstart/docker.log
 docker evens
 docker logs nms

**Comment by HackToday** [comment]
I am not sure if your base container issue or not.

Could you try one official docker image if it work or still have space issue ?

**Comment by vdemeester** [comment]
@nickmaleao isn't your *ad* mechanism mounting (network mount) stuff inside the container (like the home of one user) ? That might explains these numbers for the size.

**Comment by phemmer** [comment]
This is a common issue with using AD users inside docker. The problem is that linux creates `/var/log/wtmp

**Comment by phemmer** [comment]
This is a common issue with using AD users inside docker. The problem is that linux creates `/var/log/wtmp` as a sparse file, and the size is controlled by the user ID. Meaning for example user 0 occupies bytes 0-9, user 1 occupies bytes 10-19, user 2 uses 20-29, etc. AD users have really big numeric IDs (in the millions), meaning user 123456789 occupies bytes 1234567890-1234567899. But as mentioned, this is a sparse file, so it isn't actually using 1234567899 bytes on the filesystem. 

In short, docker is doing nothing wrong, and everything is working as it should :-)

**Comment by nickmaleao** [comment]
@phemmer , thanks for the info, you are right!

Finally a answer for this problem, but what is strange is that this problem only occurs if a try to use a ad user with ssh.

If i ssh with a local account and then change to a ad user, this problem doesn't occur.

**Comment by phemmer** [comment]
> If i ssh with a local account and then change to a ad user, this problem doesn't occur.

`/var/log/lastlogin` is updated by the `pam_lastlog` module. So the configuration of the pam stack for whatever you're using to change users doesn't invoke the `pam_lastlog` module.
